src/Controller/Component/UserToolComponent.php

```diff
@@ -365,7 +365,16 @@ public function mapAction() {
  * @return bool
  */
 	public function login($options = []) {
+		$Auth = $this->_getAuthObject();
+
 		$options = Hash::merge($this->_config['login'], $options);
+		if ($options['successRedirectUrl'] === null) {
+			$options['successRedirectUrl'] = $Auth->redirectUrl();
+		}
+
+		if ($Auth->user('id')) {
```

Comment:
You can't rely on the user `id` being present, could be a token as well for state less systems.